updater/lib/dependabot/updater/operations/create_security_update_pull_request.rb

```diff
@@ -69,6 +80,15 @@ def check_and_create_pr_with_error_handling(dependency)
             error_type: "inconsistent_registry_response",
             error_detail: e.message
           )
+        rescue StandardError
+          log_error_for_security_update_failure(dependency: dependency)
+        end
+
+        def log_error_for_security_update_failure(dependency:)
+          raise SecurityUpdateError.new(
+            message: "Security Update Error, No files were updated!",
```

Comment:
@landongrindheim , i have redone the code , earlier issue was that due to a double exception throw (`StandardError` and then `SecurityUpdateError`), there was an issue related with double log of sentry issue. so i didn't opt for a double exception.

I have raised a new exception class (`SecurityUpdateError`) and maintained the context info from thrown exception that should maintain the error_context from older exception. I have also setup the exception backtrack from `SecurityUpdateError` to thrown exception which should maintain the stack trace. It should start logging security updates error under new category.

Reply:
I think what Landon was saying is still the case, _any_ error that raises here will now be raised as a `SecurityUpdateError` instead of its native error class?